provider/datagen/src/baked_exporter.rs

```diff
@@ -282,21 +282,30 @@ impl BakedExporter {
         }
     }
 
-    fn write_impl_macro(
+    fn write_impl_macros(
         &self,
         body: TokenStream,
+        iterable_body: TokenStream,
         key: DataKey,
         marker: TokenStream,
     ) -> Result<(), DataError> {
         let marker_string = marker.to_string();
+        let marker_last = marker.into_iter().last().unwrap();
         let doc = format!(
             " Implement `DataProvider<{}>` on the given struct using the data",
-            marker.into_iter().last().unwrap()
+            marker_last
+        );
+        let doc_iterable = format!(
+            " Implement `IterableDataProvider<{}>` on the given struct using the data",
+            marker_last
         );
 
         let ident = Self::ident(key);
 
         let prefixed_macro_ident = format!("__impl_{ident}").parse::<TokenStream>().unwrap();
+        let prefixed_macro_ident_iterable = format!("__impliterable_{ident}")
```

Comment:
```suggestion
        let prefixed_macro_ident_iterable = format!("__impl_iterable_{ident}")
```

Reply:
I was considering `impl_iterable` but I went with `impliterable` so that they sort together. Note that `impl_iterable` would sort in the middle between keys starting with `h` and `j`.

https://icu4x-pr-artifacts.storage.googleapis.com/gha/ba4f909d0e8abd5228918d43b277532c525c8e04/rustdoc/icu_datetime_data/index.html

Reply:
how about `impl_{ident}_iterable`

Reply:
Then they'll get interspersed.

I think my prefs are `impliterable_{ident}` ~> `impl_{ident}_iterable` ~>> `impl_iterable_{ident}`

Reply:
yeah and I get the interspersed order changes for keys > i. Fine with whatever